lms/djangoapps/courseware/views/views.py

```diff
@@ -169,6 +172,19 @@ def courses(request):
     if not request.user.is_staff:
         courses = courses.filter(invitation_only=0)
 
+    user_messages = []
+    if latitude and longitude:
+        ordered_ccxs = sorted(ccxs, key=lambda ccx: ccx.distance_from({'latitude': latitude, 'longitude': longitude}))
+        ordered_courses = []
+
+        for ccx in ordered_ccxs:
+            if ccx.delivery_mode != CustomCourseForEdX.ONLINE_ONLY:
+                ordered_courses.extend([course for course in courses if str(course.id) == str(ccx.ccx_course_id)])
```

Comment:
Comprehension iterates thorugh whole list regardless if the result is found. Optimize.